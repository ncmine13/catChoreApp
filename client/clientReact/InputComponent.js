var React = require('react');
var ReactDOM = require('react-dom');
var NewComponent = require('./NewComponent');
var InitialView = require('./InitialView');
var request = require('superagent');


class InputComponent extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			roommatename: '',
			chorename: '',
			entryInfo: []
		}

		this.createItem = this.createItem.bind(this);
	}
	componentDidMount(){
		console.log("this is this componentDidMount", this)
		var state = this.state;
		var self = this;
		request.get('http://localhost:9292/home/chores')
		.end((err, data) => {
			state.entryInfo = data.body;
			console.log(data.body)
			self.setState(state);
		})
	}
	createItem(chore){
		console.log("this is this in create item", this)
		var state = this.state;
		var self = this;
		request.post('http://localhost:9292/home/chores')
		.type('form')
		.send(chore)
		.end((err, data) => {
			console.log(data.body, "this is data.body")
			console.log(state, "this is state within the request")
			state.data = data.body;
			self.setState(state);
			self.componentDidMount();
		})
	}
	render(){
		var chores = this.state.entryInfo;
		var reversedList = [];
		if (chores.length > 0){
				var listItems = chores.map((chore) =>
				<li key = {chore.id}>
					{chore.roommatename} {chore.chorename} on {chore.dateday}.
				</li>
			);
			reversedList = listItems.reverse();
		}		
		return (
			<div>
				<NewComponent createItem={this.createItem}/>
				{(chores.length === 0) ? <InitialView/> : <div className="listDiv"><ul className="listItems">{reversedList}</ul></div>}
			</div>
		)
	}
}


ReactDOM.render(<InputComponent/>, document.getElementById('container'));
module.exports = InputComponent;