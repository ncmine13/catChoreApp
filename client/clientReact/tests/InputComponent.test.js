import React from 'react';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json'
import InputComponent from '../InputComponent';


describe('The InputComponent', () => {
	test('should render as expected', () => {
		const component = shallow(<InputComponent />)
		const tree = toJson(component)
		expect(tree).toMatchSnapshot();
	});
});


describe('The database', () => {
	test('is empty when it contains no entries', () => {
		const chores = [];
		expect(chores.length).toEqual(0);
	});
});


// test('Request to chores API works', () => {
// 	var server = sinon.fakeServer.create();
// 	server.respondWith('get', '/chores')

// })

//post request should work
//get request should work