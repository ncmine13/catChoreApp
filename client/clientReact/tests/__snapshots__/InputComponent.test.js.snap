// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The InputComponent should render as expected 1`] = `
<div>
  <NewComponent
    createItem={[Function]}
  />
  <InitialView />
</div>
`;
